Bencodex/Types/List.cs

```diff
@@ -90,11 +97,31 @@ IEnumerator IEnumerable.GetEnumerator()
             return ((IEnumerable)Value).GetEnumerator();
         }
 
-        IImmutableList<IValue> IImmutableList<IValue>.Add(IValue value)
+        public IImmutableList<IValue> Add(IValue value)
         {
             return new List(Value.Add(value));
         }
 
+        public List Add(string value)
```

Comment:
Why the return type is `List` instead of `IImutableList<IValue>`?

Reply:
It's because `IImutableList<IValue>` interface has only the method of the signature, `IImutableList<IValue> Add(IValue)`. So it might be not available like the below case:

~~~ csharp
List.Empty
  .Add(new Text())
  .Add("text"); // ERROR: Argument type 'string' is not assignable to parameter type 'Bencodex.Types.IValue'.
~~~

I believe it will be better to return `List` but `IImutableList<IValue>.Add(IValue)` must be implemented because it is required by the interface. So, you should take care of *the order* when you use this feature.

Reply:
> I believe it will be better to return List but IImutableList<IValue> IImutableList<IValue>.Add(IValue) must be implemented because it is required by the interface. So, you should take care of the order when you use this feature.

@limebell The problem I commented, was resolved in 380e7bc. Now, even if you call `Add(IValue)`, it will return `Bencodex.Types.List` and can chain! 😃 🎉  /cc @libplanet